### PR TITLE
Add rt1180 gpt support

### PR DESCRIPTION
**Pull request by lucien-nxp**
add GPT support on RT1180

### CODE REVIEW

File: soc/nxp/imxrt/imxrt118x/soc.c [hunk 1]
```diff
@@ -233,6 +233,20 @@ static ALWAYS_INLINE void clock_init(void)
 	CLOCK_SetRootClock(kCLOCK_Root_Lpspi0102, &rootCfg);
 #endif
 
+#if defined(CONFIG_COUNTER_MCUX_GPT) && \
```
Comment (danieldegrasse) [inline soc/nxp/imxrt/imxrt118x/soc.c:236]: This should be split, into two `#if` statements, right? IE only clock GPT1 if the GPT1 nodelabel is enabled, and the same for GPT2

File: dts/arm/nxp/nxp_rt118x.dtsi [hunk 2]
```diff
@@ -90,6 +90,22 @@
 		gpio-controller;
 		#gpio-cells = <2>;
 	};
+
+	/* GPT1 is used for the hardware timer, not as a standard counter */
```
Comment (danieldegrasse) [inline dts/arm/nxp/nxp_rt118x.dtsi:94]: Is this the case here? IE has GPT1 been tested as a system clock source (which is what we use hardware timers for)

File: dts/arm/nxp/nxp_rt118x.dtsi [hunk 3]
```diff
@@ -90,6 +90,22 @@
...
+	gpt_hw_timer: gpt@46C0000 {
```
Comment (danieldegrasse) [inline dts/arm/nxp/nxp_rt118x.dtsi:95]:
```suggestion
	gpt_hw_timer: gpt@46c0000 {
		compatible = "nxp,gpt-hw-timer";
		reg = <0x46c0000 0x4000>;
```
style nit- make the alphabetic characters in the register addresses lowercase

File: dts/arm/nxp/nxp_rt118x.dtsi (hunk 2 above)
Comment (lucien-nxp) [reply dts/arm/nxp/nxp_rt118x.dtsi:94]: No, GPT1 as HW timer isn't be tested, I only refer other RT platforms. 
I have modified this gpt to GPT1, if If there is a need for HW TIMER in the future, I will change GPT1 to HW TIMER.

File: soc/nxp/imxrt/imxrt118x/soc.c (hunk 1 above)
Comment (lucien-nxp) [reply soc/nxp/imxrt/imxrt118x/soc.c:236]: Updated to split.

File: dts/arm/nxp/nxp_rt118x.dtsi (hunk 3 above)
Comment (lucien-nxp) [reply dts/arm/nxp/nxp_rt118x.dtsi]: Updated.

**Review by danieldegrasse — DISMISSED** [review]

File: soc/nxp/imxrt/imxrt118x/soc.c (hunk 1 above)
Comment (dleach02) [reply soc/nxp/imxrt/imxrt118x/soc.c:236]: Just a personal thing but I've seen this type of code structure do the following:

#if defined(CONFIG_COUNTER_MCUX_GPT)
#if (DT_NODE_HAS_STATUS(DT_NODELABEL(gpt1), okay))
/* insert code */
#endif /* DT_NODELABEL(gpt1) */

#if (DT_NODE_HAS_STATUS(DT_NODELABEL(gpt2), okay))
/* insert code */
#endif /* DT_NODELABEL(gpt2) */

#endif /* CONFIG_COUNTER_MCUX_GPT */

**Review by dleach02 — DISMISSED** [review]

File: soc/nxp/imxrt/imxrt118x/soc.c (hunk 1 above)
Comment (lucien-nxp) [reply soc/nxp/imxrt/imxrt118x/soc.c:236]: Yes, it seems better for users, I will change them like comment above. Thank you.

**Review by danieldegrasse — DISMISSED** [review]

File: boards/nxp/mimxrt1180_evk/doc/index.rst [hunk 4]
```diff
@@ -101,7 +100,7 @@ already supported, which can also be re-used on this mimxrt1180_evk board:
 +-----------+------------+-------------------------------------+
 | GPIO      | on-chip    | gpio                                |
 +-----------+------------+-------------------------------------+
-| COUNTER   | on-chip    | counter                             |
+| COUNTER   | on-chip    | gpt                                 |
```
Comment (decsny) [inline boards/nxp/mimxrt1180_evk/doc/index.rst:103]: this is backwards, left column should be GPT and right column should be counter

File: dts/arm/nxp/nxp_rt118x.dtsi
Comment (decsny) [inline dts/arm/nxp/nxp_rt118x.dtsi:1]: please combine dts and soc changes to one commit

**Review by decsny — CHANGES_REQUESTED** [review]

File: dts/arm/nxp/nxp_rt118x.dtsi
Comment (lucien-nxp) [reply dts/arm/nxp/nxp_rt118x.dtsi:1]: Squashed.
May I ask for the requirement for commit?   Before, different folder update need to split different commit, why this should be squashed when dts/soc are all changed.

Comment (decsny) [reply dts/arm/nxp/nxp_rt118x.dtsi:1]: they are just such small commits, it's easier to review the history in my opinion, it's not a requirement (the change request is just over that table)

**Review by decsny — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**